packages/core/src/capabilities/NodeInfo.ts

```diff
@@ -124,32 +124,47 @@ export function parseCCList(payload: Buffer): {
 	};
 	let offset = 0;
 	let isAfterMark = false;
-	while (offset < payload.length) {
+	let listEnd = payload.length;
+	if (isLongRange) {
+		validatePayload(payload.length >= offset + 1);
+		const listLength = payload[offset++];
+		listEnd = offset + listLength;
+		validatePayload(payload.length >= listEnd);
+	}
+	while (offset < listEnd) {
 		// Read either the normal or extended ccId
 		const { ccId: cc, bytesRead } = parseCCId(payload, offset);
 		offset += bytesRead;
 		// CCs before the support/control mark are supported
 		// CCs after the support/control mark are controlled
+		// BUGBUG: does the "mark" and support/control convention apply to isLongRange?
 		if (cc === CommandClasses["Support/Control Mark"]) {
 			isAfterMark = true;
 			continue;
 		}
 		(isAfterMark ? ret.controlledCCs : ret.supportedCCs).push(cc);
 	}
+	// BUGBUG: isLongRange prohibits CC from 0x00..0x20 from being advertised here, as does 4.3.2.1.1.17
+	// BUGBUG: how do >0xFF CC get advertised? I don't immediately see a mechanism for indicating a multi-byte CC
 	return ret;
 }
 
 export function encodeCCList(
 	supportedCCs: readonly CommandClasses[],
 	controlledCCs: readonly CommandClasses[],
+	isLongRange: boolean = false,
 ): Buffer {
-	const bufferLength =
-		sum(supportedCCs.map((cc) => (isExtendedCCId(cc) ? 2 : 1)))
+	const bufferLength = (isLongRange ? 1 : 0)
+		+ sum(supportedCCs.map((cc) => (isExtendedCCId(cc) ? 2 : 1)))
 		+ (controlledCCs.length > 0 ? 1 : 0) // support/control mark
 		+ sum(controlledCCs.map((cc) => (isExtendedCCId(cc) ? 2 : 1)));
 
 	const ret = Buffer.allocUnsafe(bufferLength);
 	let offset = 0;
+	if (isLongRange) {
+		// BUGBUG: validate bufferLength - 1 is <= 0xFF
```

Comment:
If this length can't be encoded in a byte, what's an appropriate error action here? Throw an error? Other?